### PR TITLE
gdrive remote service accounts can use env var

### PR DESCRIPTION
**Pull request by shcheklein**
For service accounts it is also an option to pass keys via an env variable.

> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: content/docs/user-guide/setup-google-drive-remote.md [hunk 1]
```diff
@@ -195,6 +195,13 @@ authentication is needed.
                  gdrive_service_account_json_file_path path/to/file.json
    ```
 
+   Alternatively, a `GDRIVE_CREDENTIALS_DATA` can be set to pass service account
+   key in CI/CD systems, production setup, read-only file systems, etc. The
+   content of this variable should be a string with JSON that has the same
+   format as in the keys file described above. If both this variable and
+   `gdrive_service_account_json_file_path` are provide, DVC reads the
+   environment variable first.
```
Comment (daavoo) [inline content/docs/user-guide/setup-google-drive-remote.md:203]: I think that providing `gdrive_service_account_json_file_path` is no longer required after https://github.com/iterative/dvc/pull/7213

Comment (daavoo) [reply content/docs/user-guide/setup-google-drive-remote.md:203]: > I read the section as saying that either gdrive_service_account_json_file_path or GDRIVE_CREDENTIALS_DATA is required.

I understood the phase wrong 🙏

Comment (shcheklein) [reply content/docs/user-guide/setup-google-drive-remote.md]: @dberenbaum @daavoo thanks! Done, I've changed the message a bit - pls take a look.

Comment (shcheklein) [reply content/docs/user-guide/setup-google-drive-remote.md]: > Finally, should we promote GDRIVE_CREDENTIALS_DATA as the primary method 

Make sense, but let's keep it as is for now.  Service accounts can be used (and are used) locally as well ...

File: content/docs/user-guide/setup-google-drive-remote.md [hunk 2]
```diff
@@ -195,6 +195,14 @@ authentication is needed.
                  gdrive_service_account_json_file_path path/to/file.json
    ```
 
+   Alternatively, a `GDRIVE_CREDENTIALS_DATA` can be set to pass service account
+   key in CI/CD systems, production setup, read-only file systems, etc. The
+   content of this variable should be a string with JSON that has the same
+   format as in the keys file described above. If both this variable and
+   `gdrive_service_account_json_file_path` are provide,
```
Comment (dberenbaum) [inline content/docs/user-guide/setup-google-drive-remote.md:202]:
```suggestion
   `gdrive_service_account_json_file_path` are provided,
```

**Comment by dberenbaum** [comment]
One more thought that doesn't need to block this PR: should the "Authorization" section be moved above "Using service accounts"? I find it confusing to go through a section about service accounts and then have a section after it focused back on user accounts without being clear that it's a different use case from the above section.

**Comment by shcheklein** [comment]
> One more thought that doesn't need to block this PR

Yep, it's a good point also, but let me do this a separate PR though.

**Comment by shcheklein** [comment]
@jorgeorpinel @iesahin ping :)

**Review by daavoo — APPROVED** [review]

**Comment by shcheklein** [comment]
@dberenbaum here is the followup https://github.com/iterative/dvc.org/pull/3473 :)

**Pull request merged**